processor/k8sprocessor/processor.go

```diff
@@ -92,56 +92,31 @@ func (kp *kubernetesprocessor) ProcessMetrics(ctx context.Context, md pdata.Metr
 func (kp *kubernetesprocessor) ProcessLogs(ctx context.Context, ld pdata.Logs) (pdata.Logs, error) {
 	rl := ld.ResourceLogs()
 	for i := 0; i < rl.Len(); i++ {
-		kp.processResource(ctx, rl.At(i).Resource(), k8sIPFromAttributes())
+		kp.processResource(ctx, rl.At(i).Resource())
 	}
 
 	return ld, nil
 }
 
-func (kp *kubernetesprocessor) processResource(ctx context.Context, resource pdata.Resource, attributeExtractors ...ipExtractor) {
-	var podIP string
+// processResource adds Pod metadata tags to resource based on pod association configuration
+func (kp *kubernetesprocessor) processResource(ctx context.Context, resource pdata.Resource) {
 
-	for _, extractor := range attributeExtractors {
-		podIP = extractor(resource.Attributes())
-		if podIP != "" {
-			break
-		}
-	}
-
-	// Check if the receiver detected client IP.
-	if podIP == "" {
-		if c, ok := client.FromContext(ctx); ok {
-			podIP = c.IP
-		}
-	}
-
-	// If IP is still not available by this point, nothing can be tagged here. Return.
-	if podIP == "" {
+	podIdentifierLabel, podIdentifierValue := extractPodIds(ctx, resource.Attributes(), kp.podAssociations)
```

Comment:
I think it is best not to use the term "label" in this context. OpenTelemetry's vocabulary for Resources is "attribute name" or "attribute key". I think the following would be clear:
```suggestion
	podIdentifierKey, podIdentifierValue := extractPodIds(ctx, resource.Attributes(), kp.podAssociations)
```